test/libsolidity/smtCheckerTests/types/address_transfer_3.sol

```diff
@@ -0,0 +1,15 @@
+contract C
+{
+	function f(address payable a) public {
+		require(1000 == address(this).balance);
+		require(100 == a.balance);
```

Comment:
This example is nice, the solver infers that `a == this` is not possible because the balances are different.